### DISCUSSION TITLE
Multitenancy

### DISCUSSION THREAD

**Discussion by scottlet** [nextauthjs/next-auth#10090] [Help]
I have two websites - they’re actually just the one website but using cnames. One serving .co.uk and one serving .ie
I am currently using OidcClient for auth and I am trying to move our auth to NextAuth
But I am stuck on the `NEXTAUTH_URL` having to be an environment variable, when I need it to be either `.ie/order/api/auth` or `.co.uk/order/api/auth` depending on which website the original request came from.

I cannot find a sensible solution to this, does anyone have any suggestions? About the only way I can think of right now is to actually split this into two website deployments, doubling my infrastructure costs, but allowing separate environment variables to exist for each tenant.

**Comment by scottlet** [comment]
...would `redirectProxyUrl` and lazy initialisation in v5 allow me to do the above? Changing the redirect URL to either .co.uk or .ie depending on the host?

**Comment by MarkLyck** [comment]
@scottlet Did you ever figure this out? We're also struggling getting multi tenancy to work with next-auth@5

**Reply by scottlet** [reply]
I did not. As we already had a few pods serving the site it made more sense for us to split our infra between IE and UK than wait for this change to land as it was almost the same cost to us. So we have dedicated environment variables for IE and UK now.
